mono/metadata/boehm-gc.c

```diff
@@ -34,6 +34,11 @@
 #include <mono/utils/mono-os-mutex.h>
 #include <mono/utils/mono-counters.h>
 
+#ifdef _MSC_VER
+	// Quiet Visual Studio linker warning, LNK4221, in cases when this source file intentional ends up empty.
+	void __mono_win32_boehm_gc_quiet_lnk4221(void) {}
```

Comment:
Yes, you could of course turn of the linker warning for the complete project, but that will not be as fine grained as the above solution, since it will ignore the warning across all objects and not for specific individual objects as done with current solution. There was also a different idea to just suppress the warning in cases where the source file intentionally ends up empty, so in other scenarios (the unintentionally case), the warning will still be issued for these files (after I fixed up the placement, stay tuned).

I see that there is currently a compile error when building the bohem case of this PR, I will look into that and move the code to only be used when, for example, the boehm gc is not used, since that should only be the case when the source file should be empty in the first place. Other scenarios could then still trigger the warning if still ending up with an unintentional empty source file. What do you think about that?